### PR TITLE
[DF] Add GetDataSetInfo function

### LINKED ISSUE #7829: [DF] Introduce GetDatasetInfo and add info to Describe
The idea is to have an identifier that describes the used dataset (name of the TTree or similar), make this info accessible as public API and also add it to `Describe`. See the discussion here:

https://github.com/root-project/root/issues/7561#issuecomment-817629082

https://github.com/root-project/root/issues/7561#issuecomment-817717882

Remember to make the identifier returned by this API subject to change because it's for interactive use and not to be parsed automatically.

### PR DESCRIPTION
**Pull request by stwunsch**
Fixes #7829 partially, missing integration into Describe

### CODE REVIEW

**Comment by eguiraud** [comment]
As per our discussion we'd like this method to report a bit more, namely anything that is useful to answer "what data is this RDF running on?", so e.g. both tree name and file names should be mentioned

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 1]
```diff
@@ -2191,6 +2193,97 @@ public:
    /// ~~~
    unsigned int GetNRuns() const { return fLoopManager->GetNRuns(); }
 
+   /// \brief Gets descriptive information about the dataset
```
Comment (eguiraud) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2196]:
```suggestion
   /// \brief Get descriptive information about the dataset.
```

File: tree/dataframe/inc/ROOT/RDF/RInterface.hxx [hunk 2]
```diff
@@ -2191,6 +2193,97 @@ public:
...
+   /// \return Info describing the dataset
```
Comment (eguiraud) [inline tree/dataframe/inc/ROOT/RDF/RInterface.hxx:2197]:
```suggestion
   /// \return Info describing the dataset as a multi-line string
```

**Review by eguiraud — APPROVED** [review]
Looks great, thanks @stwunsch !

**Comment by stwunsch** [comment]
Note to myself: Add this to the RDF cheat sheet and the release notes.

**Pull request merged**